yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -66,14 +66,16 @@ private[spark] class ApplicationMaster(
   // allocation is enabled), with a minimum of 3.
 
   private val maxNumExecutorFailures = {
-    val defaultKey =
+    val effectiveNumExecutors =
```

Comment:
This does more than back-port the fix. It actually back-ports the behavior change that required the fix. I'm not clear we want to do that? that is, the default is now very different but I'm not sure what other changes would be necessary to make that correct in this branch.

If it's not fixing the same problem, I'd leave this alone.